### ISSUE TITLE
Convert to static site

### ISSUE THREAD

**Issue by fmaclen** [fmaclen/hollama#1]
- [ ] Save session/contex on browser's `localStorage`
- [ ] Add ability to set the Ollama server and model in the UI
- [x] Update config to deploy to Cloudflare Pages

**Comment by cipriancraciun** [comment]
Sorry for touching on this closed issue, but although the tool does seem to support static serving (I've done so by downloading the Electron app, extracting the ASAR, and obtaining the `index.html` from the online version), this fact is not publicly advertised.

Here a few items that might help others achieving this:
* point-out in the README that this UI is also available for static hosting;  (**this is a major feature, as most don't want to run yet another service on their infrastructure**;)  (it does mention once that "this is a static site built with", but nothing more than that;)
* provide a snippet that allows one to build the required static files;
* provide as a release an archive containing the static files required for hosting;

**Comment by fmaclen** [comment]
@cipriancraciun I'm a bit confused about what you are looking at, there shouldn't be an `index.html` in the extracted Electron app, this is what the unpacked `app.asar` looks like in version 0.25.1:

![Image](https://github.com/user-attachments/assets/f1bc5b56-9913-40c1-8549-6dd6a2505e79)

Unfortunately, the current version of Hollama needs to be run as a server. Even the Electron app [spins up a server](https://github.com/fmaclen/hollama/blob/main/electron/main.js#L57-L62).

If you don't want to setup your own infra you can use the public server at https://hollama.fernando.is, it only uses the cloud to download the latest version of the UI, but all of the data and interactions are handled locally on your browser.

**Comment by cipriancraciun** [comment]
> I'm a bit confused about what you are looking at, there shouldn't be an index.html in the extracted Electron app. [...]

Indeed, the `app.asar` doesn't contain an `index.html`, however one could obtain a rendered one from <https://hollama.fernando.is/> with a simple `curl`.

> Unfortunately, the current version of Hollama needs to be run as a server. [...]

Fortunately (if I remember correctly, because I've tried multiple Ollama frontends), if one places that `index.html` inside the same folder with `_app` that is extracted from `app.asar`, and then serves those on HTTP with a plain static server, basic functionality works without a problem (i.e. listing models, interacting with a chosen model).

It might perhaps lack some advanced features that rely on the electron server, or the hosted server, but at least basic functionality just works.  (And, if it happens to use local browser storage for local data, then I see no reason why one can't just serve the static files.)
